pyccel/codegen/printing/ccode.py

```diff
@@ -104,6 +109,82 @@ def _traverse_matrix_indices(self, mat):
     def _print_Module(self, expr):
         return '\n\n'.join(self._print(i) for i in expr.body)
 
+    def _print_While(self,expr):
+        code = "while (%s)\n{" % self._print(expr.test)
+        code = code + "\n %s" % self._print(expr.body) + "\n}"
+        return (code)
+
+    def _print_If(self, expr):
+        lines = []
+        for i, (c, e) in enumerate(expr.args):
+            var = self._print(e)
+            #change this line (removes an extra empty else{})
+            if (var == ''):
+                break
+            if i == 0:
+                lines.append("if (%s)\n{" % self._print(c))
+            elif i == len(expr.args) - 1 and c is BooleanTrue():
+                lines.append("else\n{")
+            else:
+                lines.append("else if (%s)\n{" % self._print(c))
+            #e can be a tuple when one python statement
+            #results in multiple c statements
+            if isinstance(e, (list, tuple, Tuple, PythonTuple)):
```

Comment:
Are you sure that `e` can be a `PythonTuple`? A `PythonTuple` is used to represent a tuple which was written in python, whereas the other 3 are used to collect statements

Reply:
I think e can't be neither of those types, there is no need for that condition to exist.
I was thinking e could be a tuple when a python statement results in multiple c statements, but this isn't the case. I tested it and e is always of type `CodeBlock`.
Same goes for fortran

Reply:
Ah, very good. I though that ought to be the case. In that case this if loop can be removed